packages/strapi-plugin-content-type-builder/admin/src/components/ModalHeader/index.js

```diff
@@ -1,89 +1,109 @@
 import PropTypes from 'prop-types';
 import { HeaderModalTitle } from 'strapi-helper-plugin';
 import { get } from 'lodash';
 import { AttributeIcon } from '@buffetjs/core';
 import { FormattedMessage } from 'react-intl';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import useDataManager from '../../hooks/useDataManager';
-import pluginId from '../../pluginId';
 import ComponentIcon from './ComponentIcon';
-import Header from './Header';
+import ComponentInfos from './ComponentInfos';
+import IconWrapper from './IconWrapper';
+import UpperFirst from '../UpperFirst';
+import DropdownInfos from './DropdownInfos';
 
-const ModalHeader = ({
-  category,
-  headerId,
-  iconType,
-  name,
-  target,
-  targetUid,
-  subCategory,
-  subTargetUid,
-}) => {
-  const { modifiedData } = useDataManager();
-  const currentComponent = get(modifiedData, ['components', targetUid], {});
-  const shouldDisplayComponentCatInfos = target === 'components';
-  const currentComponentIcon = get(currentComponent, ['schema', 'icon'], '');
-
-  let iconName;
-
-  if (iconType === 'components') {
-    iconName = 'component';
-  } else {
-    iconName = iconType;
-  }
+const ModalHeader = ({ headerId, headers }) => {
+  const shouldDisplayDrodDrown = headers.length > 3;
 
   return (
     <section>
       <HeaderModalTitle style={{ textTransform: 'none' }}>
-        {shouldDisplayComponentCatInfos ? (
-          <ComponentIcon isSelected>
-            <FontAwesomeIcon icon={currentComponentIcon} />
-          </ComponentIcon>
-        ) : (
-          <AttributeIcon
-            type={iconName}
-            style={{ margin: 'auto 20px auto 0' }}
-          />
-        )}
         {headerId && (
-          <FormattedMessage id={`${pluginId}.${headerId}`} values={{ name }} />
-        )}
-        {!headerId && (
-          <Header
-            category={category}
-            name={name}
-            target={target}
-            targetUid={targetUid}
-            subCategory={subCategory}
-            subTargetUid={subTargetUid}
-          />
+          <>
+            <AttributeIcon
+              type={get(headers, [0, 'icon', 'name'], '')}
+              style={{ margin: 'auto 20px auto 0' }}
+            />
+            <FormattedMessage
+              id={headerId}
+              values={{ name: get(headers, [0, 'label'], '') }}
+            />
+          </>
         )}
+        {!headerId &&
+          headers.map((header, index) => {
+            const iconName = get(header, ['icon', 'name'], '');
+            const iconType = iconName === null ? '' : iconName;
+            const icon = get(header, ['icon', 'isCustom'], false) ? (
+              <ComponentIcon isSelected>
+                <FontAwesomeIcon icon={iconType} />
+              </ComponentIcon>
+            ) : (
+              <AttributeIcon
+                type={iconType}
+                style={{ margin: 'auto 20px auto 0' }}
+              />
```

Comment:
You use the AttributeIcon component third time with the same style, maybe it can be nice to use styled-component to override this style, as you did for `Dropdown`, `DropdownToggle`, `DropdownMenu` etc…